### DISCUSSION TITLE
What node type contains data in RAM? How do I deploy data nodes and storage nodes on different servers?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#19157] [Q&A and General discussion]
What node type contains data in RAM? How do I deploy data nodes and storage nodes on different servers?

**Comment by yhmo** [comment]
There are three kinds of working nodes:
1. query node, caches segments in RAM for searching/querying. It requires large memory size, depends how much data need to be searched.
2. data node, subscribes data from Pulsar and persist segments into MinIO. Typically it requires 2 ~ 3 times of a segment size.
3. index node, build index for segment. It requires 2 ~ 3 times of a segment size.
